autopush/tests/test_integration.py

```diff
@@ -100,8 +100,11 @@ def __getattribute__(self, name):
         else:
             return f
 
-    def connect(self):
-        self.ws = websocket.create_connection(self.url)
+    def connect(self, connection_port=None):
+        url = self.url
+        if connection_port:
+            url = "ws://localhost:{}/".format(connection_port)
```

Comment:
This is pretty meh, how about
```python
def reconnect(url):
    if self.ws.connected:
        self.disconnect()
    self.url = url
    return self.connect()
```

easier than disconnect(); connect(); and also works when you need to do something in between those 2

Reply:
That doesn't check that the current connection is to the given URL, not sure how easy it is to determine that.

Reply:
where do you need that?

Reply:
It's also not a reconnect if its connecting to a different URL than what it was on, that's a new connect. But now its just semantics,sigh.